storage/aptosdb/src/pruner/ledger_pruner_worker.rs

```diff
@@ -1,93 +1,65 @@
 /// This also exposes API to report the progress to the parent thread.
+#[derive(Debug)]
 pub struct LedgerPrunerWorker {
-    command_receiver: Receiver<db_pruner::Command>,
+    /// The worker will sleep for this period of time after pruning each batch.
+    pruning_time_interval_in_ms: u64,
     /// Ledger pruner.
     pruner: Arc<LedgerPruner>,
-    /// Indicates if there's NOT any pending work to do currently, to hint
-    /// `Self::receive_commands()` to `recv()` blocking-ly.
-    blocking_recv: bool,
     /// Max items to prune per batch. For the ledger pruner, this means the max versions to prune
     /// and for the state pruner, this means the max stale nodes to prune.
-    ledger_store_max_versions_to_prune_per_batch: u64,
+    max_versions_to_prune_per_batch: u64,
+    /// Indicates whether the pruning loop should be running. Will only be set to true on pruner
+    /// destruction.
+    disable_pruner: AtomicBool,
 }
 
 impl LedgerPrunerWorker {
     pub(crate) fn new(
         ledger_pruner: Arc<LedgerPruner>,
-        command_receiver: Receiver<db_pruner::Command>,
         storage_pruner_config: StoragePrunerConfig,
     ) -> Self {
         Self {
+            pruning_time_interval_in_ms: storage_pruner_config.ledger_pruner_time_interval_in_ms,
             pruner: ledger_pruner,
-            command_receiver,
-            blocking_recv: true,
-            ledger_store_max_versions_to_prune_per_batch: storage_pruner_config
-                .ledger_pruning_batch_size
-                as u64,
+            max_versions_to_prune_per_batch: storage_pruner_config.ledger_pruning_batch_size as u64,
+            disable_pruner: AtomicBool::new(false),
         }
     }
 
-    pub(crate) fn work(mut self) {
-        while self.receive_commands() {
-            // Process a reasonably small batch of work before trying to receive commands again,
-            // in case `Command::Quit` is received (that's when we should quit.)
-            let mut error_in_pruning = false;
-
-            self.pruner
-                .prune(self.ledger_store_max_versions_to_prune_per_batch as usize)
-                .map_err(|_| error_in_pruning = true)
-                .ok();
-
-            if !self.pruner.is_pruning_pending() || error_in_pruning {
-                self.blocking_recv = true;
-            } else {
-                self.blocking_recv = false;
+    // Loop that does the real pruning job.
+    pub(crate) fn work(&self) {
+        while !self.disable_pruner.load(Ordering::Relaxed) {
+            let pruner_result = self
+                .pruner
+                .prune(self.max_versions_to_prune_per_batch as usize);
+            if pruner_result.is_err() {
+                println!(
+                    "Ledger pruner has error: {:?}",
+                    pruner_result.err().unwrap()
+                )
             }
+            sleep(Duration::from_millis(self.pruning_time_interval_in_ms));
```

Comment:
Should only sleep when no big enough batch to work on?

Reply:
Done